### ISSUE TITLE
Add a TestResult "id" property

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axrl#7]
Feedback from Peter Durham:

It would be helpful to have the "id" for each rule (we use for tags and stats)

**Comment by WilcoFiers** [comment]
I agree. My initial thinking was to rely on the `helpUrl` as a way to identify what rule/test/requirement was run, but having an identifier that persists between versions is useful I think.

I do not think this should be called "id". Maybe something like "test"? I would like it to be able to express results from testing WCAG success criteria as well. I think [EARL's test property](https://www.w3.org/TR/EARL10-Schema/#test) is basically what I'm after.

**Comment by AutoSponge** [comment]
Looking at https://schema.org/MedicalTest, has `code:MedicalCode`, `guideline:MedicalGuideline`, `recognizingAuthority:Organization` which I think map well to various codes and guidelines we have from WCAG, 508, AODA, ISO, etc.
